### PR TITLE
Tier2 : NFS new automation for post upgrade CephFS validation

### PR DESCRIPTION
**Pull request by sumabai**
# Description
CEPH-83575098 - Upgrade from RHCS n-1 version to latest n version with existing stand-alone NFS cluster with exports and ensure that the older exports are functioning as expected by continuously writing IOs, create more exports after upgrade and validate

Logs: Validated Post upgrade NFS test automation across below suites,

    1.tier-0_cephfs_upgrade_5x_to_7x. :  Logs - http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-D3Z7IY
    2. tier-0_cephfs_upgrade_6x_to_7x : Logs- http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-GTSNVJ/

Logs with review comments addressed: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-OJCZ07

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

File: tests/cephfs/cephfs_upgrade/cephfs_post_upgrade_validation.py [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+log = Log(__name__)
+
+
+def nfs_test(nfs_req_params):
+    """
+    NFS post upgrade validation
+
+    Args:
+        nfs_req_params: mandatory params for testing in below format
```
Comment (neha-gangadhar) [inline tests/cephfs/cephfs_upgrade/cephfs_post_upgrade_validation.py:20]: please add parameters type to be passed while using the function like string, list etc as standard practice.

File: tests/cephfs/cephfs_upgrade/cephfs_post_upgrade_validation.py [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+        raise CommandFailed("Failed to create nfs export")
+
+    time.sleep(2)
+    log.info("Mount ceph nfs new exports and run IO")
+    nfs_mounting_dir = f"{existing_nfs_mount}_new2"
+    run_io(nfs_client, nfs_server_name_1, new_export_name, nfs_mounting_dir, fs_util)
+    log.info("NFS post upgrade validation succeeded")
+
+
+def run_io(nfs_client, nfs_server, export_name, mount_path, fs_lib):
```
Comment (neha-gangadhar) [inline tests/cephfs/cephfs_upgrade/cephfs_post_upgrade_validation.py:116]: please move the function run_io to cephfs_utilsV1.py for reuse in other tests.

**Comment by sumabai** [comment]
Neha, i had addressed your comment to add run_io to fsutil for reuse. I renamed the module as i noticed similar module already in fsutils testlib. Renamed to nfs_mount_and_io and appended, this still serves the need within fsutils testlib.

**Comment by neha-gangadhar** [comment]
@sumabai Kindly rebase the commit.

**Review by AmarnatReddy — COMMENTED** [review]
LGTM

**Review by hkadam134 — DISMISSED** [review]
LGTM

**Review by neha-gangadhar — APPROVED** [review]
lgtm

**Pull request merged**